packages/react-aria-components/docs/advanced.mdx

```diff
@@ -115,6 +115,46 @@ function Stepper({children}) {
 
 The slots provided by each built-in React Aria component are shown in the Anatomy section of their documentation.
 
+#### Default slot
+
+The default slot is used to provide props to a component without specifying a slot name. This approach allows you to assign a default slot to a component for its default use case and enables you to specify a slot name for a specific use case.
+
+This example shows a custom `Button` component with slots for an icon with no slot name or an icon with a slot named "end".
+
+```tsx
+import {Button, DEFAULT_SLOT} from 'react-aria-components';
+
+function MyCustomButton({children}) {
+  return (
+    <IconContext.Provider
```

Comment:
Should we use `ButtonContext` here in the example instead of a Icon example? I'm just concerned that people will think this would work out of the box but they would need to make their icons call `useSlottedContext`/`useContextProps` for this to work

Reply:
```
import {Button, ButtonContext, DEFAULT_SLOT} from 'react-aria-components';

function MyCustomComponent({children}) {
  return (
    <ButtonContext.Provider
      value={{
        slots: {
          [DEFAULT_SLOT]: {
            className: "left-button"
          },
          end: {
            className: "right-button"
          }
        }
      }}>
        <div>{children}</div>
    </IconContext.Provider>
  );
}

<MyCustomComponent>
  {/* Consumes the props passed to the default slot */}
  <Button>Click me</Button>
</MyCustomComponent>

<MyCustomComponent>
  {/* Consumes the props passed to the "end" slot */}
  <Button slot="end">Click me</Button>
</MyCustomComponent>
```

Would something like this be better? 

I tought about avoiding a custom icon component, but couldn't find a suitable "real" use case example that showcase how useful the DEFAULT_SLOT is.

Reply:
I think the above is fine, or you could have the default slot provide a base className/style object/etc.

Reply:
Updated the code to use the example above!